File: 2026-08-09-java-http-server/tests/TestHelper.java
import java.util.Objects;

public class TestHelper {
    static int passed = 0;
    static int failed = 0;

    public static void assertEquals(Object expected, Object actual, String message) {
        if (Objects.equals(expected, actual)) {
            passed++;
        } else {
            failed++;
            System.out.println("FAIL: " + message + "\n  expected: " + expected + "\n  actual:   " + actual);
        }
    }

    public static void assertTrue(boolean condition, String message) {
        if (condition) {
            passed++;
        } else {
            failed++;
            System.out.println("FAIL: " + message);
        }
    }

    public static void assertNull(Object value, String message) {
        assertTrue(value == null, message + " (was " + value + ")");
    }
}

File: 2026-08-11-csharp-rpg-game/tests/TestHelper.cs
namespace RpgGame.Tests;

public static class TestHelper
{
    public static int Passed;
    public static int Failed;

    public static void AssertEquals<T>(T expected, T actual, string message)
    {
        if (Equals(expected, actual))
        {
            Passed++;
        }
        else
        {
            Failed++;
            Console.WriteLine($"FAIL: {message}\n  expected: {expected}\n  actual:   {actual}");
        }
    }

    public static void AssertTrue(bool condition, string message)
    {
        if (condition)
        {
            Passed++;
        }
        else
        {
            Failed++;
            Console.WriteLine($"FAIL: {message}");
        }
    }

    public static void AssertFalse(bool condition, string message) => AssertTrue(!condition, message);
}
